tangelo/algorithms/classical/mp2_solver.py

```diff
@@ -83,7 +83,17 @@ def simulate(self):
         if self.uhf:
             self.mp2_fragment = self.mp.UMP2(self.mean_field, frozen=self.frozen)
         else:
-            self.mp2_fragment = self.mp.RMP2(self.mean_field, frozen=self.frozen)
+            import pyscf
+            if pyscf.__version__ == '2.5.0' and self.mean_field.istype('ROHF'):
+                mf = self.mean_field
+                mf = mf.remove_soscf()
+                mf = mf.to_uhf()
+                if getattr(mf, 'with_df', None):
+                    self.mp2_fragment = self.mp.UMP2(mf, frozen=self.frozen, mo_coeff=mf.mo_coeff, mo_occ=None)
+                else:
+                    self.mp2_fragment = self.mp.UMP2(mf, frozen=self.frozen, mo_coeff=mf.mo_coeff, mo_occ=None)
```

Comment:
This looks exactly like L92 to me, I am a bit confused why we have that conditional statement

Reply:
Yeshh they are very similar I took it from the pyscf code and placed it , wanted to ask about this, I don't think the result is any different for both case

Reply:
If these lines are exactly the same, let's just remove L91-93 and keep L94. If everything goes well, by the time we get a version of pyscf with the fix, this will all be trivial :)

Reply:
Will fix this